[Bug]: `fib/test_fib.py::test_hash[ipv6]` fails on `hash_key` = `ip-proto` for `255`

### Issue Description

`fib/test_fib.py::test_hash[ipv6]` fails with `hash_key=ip-proto` when it uses `255`.
```
======================================================================
FAIL: hash_test.HashTest
----------------------------------------------------------------------
Traceback (most recent call last):
  File "ptftests/py3/hash_test.py", line 558, in runTest
    self.check_hash(hash_key)
  File "ptftests/py3/hash_test.py", line 209, in check_hash
    hash_key, src_port, dst_ip, exp_port_lists)
  File "ptftests/py3/hash_test.py", line 224, in check_ip_route
    hash_key, src_port, dst_port_lists)
  File "ptftests/py3/hash_test.py", line 426, in check_ipv6_route
    self, masked_exp_pkt, dst_ports, timeout=1)
  File "/root/env-python3/lib/python3.7/site-packages/ptf/testutils.py", line 3444, in verify_packet_any_port
    % (ports, device_number, result.format())
AssertionError: Did not receive expected packet on any of ports [0, 1, 2, 3, 4, 5] for device 0.
========== EXPECTED ==========
Mask:

packet status: OK
packet:
0000  00 01 02 03 04 05 00 06 07 08 09 0A 86 DD 60 00  ..............`.
0010  00 00 00 2E FF 3F 20 D0 A8 00 00 00 00 00 00 00  .....? .........
0020  00 00 00 00 00 00 20 D0 A8 00 00 00 00 01 00 00  ...... .........
0030  00 00 00 00 00 00 04 D2 00 50 00 00 00 00 00 00  .........P......
0040  00 00 50 02 20 00 81 8D 00 00 44 44 44 44 44 44  ..P. .....DDDDDD
0050  44 44 44 44 44 44 44 44 44 44 44 44 44 44 44 44  DDDDDDDDDDDDDDDD
0060  44 44 44 44                                      DDDD
```

I reproduced this by running the ptf script again:
```
/root/env-python3/bin/ptf --test-dir ptftests/py3 hash_test.HashTest --platform-dir ptftests --qlen=20000 --platform remote -t "fib_info_files=['/root/fib_info_all_duts.txt'];ptf_test_port_map='/root/ptf_test_port_map.json';hash_keys
=['src-ip', 'dst-ip', 'src-port', 'dst-port', 'ip-proto'];src_ip_range='20D0:A800:0:00::,20D0:FFFF:0:00::FFFF';dst_ip_range='20D0:A800:0:01::,20D0:FFFF:0:01::FFFF';vlan_ids=[1032, 1033, 1034, 1035, 1036, 1037, 1038, 1039, 1040, 1041, 1042, 1043, 1044, 10
45, 1046, 1047, 1048, 1049, 1050, 1051, 1052, 1053, 1054, 1055, 1056, 1057, 1058, 1059, 1060, 1061, 1062, 1063, 1064, 1065, 1066, 1067, 1068, 1069, 1070, 1071, 1072, 1073, 1074, 1075, 1076, 1077, 1078, 1079, 1080, 1081, 1082, 1083, 1084, 1085, 1086, 1087
, 1088, 1089, 1090, 1091, 1092, 1093, 1094, 1095, 1096, 1097, 1098, 1099, 1100, 1101, 1102, 1103, 1104, 1105, 1106, 1107, 1108, 1109, 1110, 1111, 1112, 1113, 1114, 1115, 1116, 1117, 1118, 1119, 1120, 1121, 1122, 1123, 1124, 1125, 1126, 1127, 1128, 1129,
1130, 1131, 1132, 1133, 1134, 1135, 1136, 1137, 1138, 1139, 1140, 1141, 1142, 1143, 1144, 1145, 1146, 1147, 1148, 1149, 1150, 1151, 1152, 1153, 1154, 1155, 1156, 1157, 1158, 1159, 1160, 1161, 1162, 1163, 1164, 1165, 1166, 1167, 1168, 1169, 1170, 1171, 11
72, 1173, 1174, 1175, 1176, 1177, 1178, 1179, 1180, 1181, 1182, 1183, 1184, 1185, 1186, 1187, 1188, 1189, 1190, 1191, 1192, 1193, 1194, 1195, 1196, 1197, 1198, 1199, 1200, 1201, 1202, 1203, 1204, 1205, 1206, 1207, 1208, 1209, 1210, 1211, 1212, 1213, 1214
, 1215, 1216, 1217, 1218, 1219, 1220, 1221, 1222, 1223, 1224, 1225, 1226, 1227, 1228, 1229, 1230, 1231, 1232, 1233, 1234, 1235, 1236, 1237, 1238, 1239, 1240, 1241, 1242, 1243, 1244
, 1245, 1246, 1247, 1248, 1249, 1250, 1251, 1252, 1253, 1254, 1255, 1256, 1257, 1258, 1259, 1260, 1261, 1262, 1263, 1264, 1265, 1266, 1267, 1268, 1269, 1270, 1271, 1272, 1273, 1274, 1275, 1276, 1277, 1278];ignore_ttl=True;single_fib_for_duts='single-fib-
single-hop';switch_type='voq';is_active_active_dualtor=False" --relax --debug info --log-file /tmp/hash_test.HashTest.ipv6.2024-12-02-22:03:55.log --socket-recv-size 16384

(Pdb) p hash_key                                                                                                                                                                                                                                              'ip-proto'
(Pdb) p ip_proto
255
```

If I send the packet again:
```
(Pdb) send_packet( self, src_port, pkt, count=100 )
10000
```

I can see the packet sent as:
```
16:39:46.124644 52:67:2c:8e:a0:09 (oui Unknown) > 94:8e:d3:1f:b0:fd (oui Unknown), ethertype IPv6 (0x86dd), length 100: 20d0:a800:: > 20d0:a800:0:1::: ip-proto-255 46
        0x0000:  948e d31f b0fd 5267 2c8e a009 86dd 6000
        0x0010:  0000 002e ff40 20d0 a800 0000 0000 0000
        0x0020:  0000 0000 0000 20d0 a800 0000 0001 0000
        0x0030:  0000 0000 0000 04d2 0050 0000 0000 0000
        0x0040:  0000 5002 2000 818d 0000 4444 4444 4444
        0x0050:  4444 4444 4444 4444 4444 4444 4444 4444
        0x0060:  4444 4444
```

And returned from the dut as:
```
16:38:00.797992 94:8e:d3:5e:af:13 (oui Unknown) > 1e:bf:59:4c:90:6f (oui Unknown), ethertype IPv6 (0x86dd), length 96: truncated-ip6 - 4 bytes missing!20d0:a800:: > 20d0:a800:0:1::: ip-proto-255 46
        0x0000:  1ebf 594c 906f 948e d35e af13 86dd 6000
        0x0010:  0000 002e ff3f 20d0 a800 0000 0000 0000
        0x0020:  0000 0000 0000 20d0 a800 0000 0001 0000
        0x0030:  0000 0000 0000 0000 0000 0000 0000 5002
        0x0040:  2000 818d 0000 4444 4444 4444 4444 4444
        0x0050:  4444 4444 4444 4444 4444 4444 4444 4444
```

You can see that the tcp-sport (04d2) and tcp-dport (0050) are removed.

### Results you see

`fib/test_fib.py::test_hash[ipv6]` fails with `hash_key=ip-proto`

### Results you expected to see

`fib/test_fib.py::test_hash[ipv6]` should pass with `hash_key=ip-proto`

### Is it platform specific

generic

### Relevant log output

_No response_

### Output of `show version`

_No response_

### Attach files (if any)

_No response_

--- a/ansible/roles/test/files/ptftests/py3/hash_test.py
+++ b/ansible/roles/test/files/ptftests/py3/hash_test.py
@@ -242,7 +242,8 @@ class HashTest(BaseTest):
         # ip_proto 60 is redirected to ip_proto 4 as encapsulation protocol, ip payload will be malformat
         # ip_proto 254 is experimental
         # MLNX ASIC can't forward ip_proto 254, BRCM is OK, skip for all for simplicity
-        skip_protos = [2, 253, 4, 41, 60, 254]
+        # ip_proto 255 is reserved
+        skip_protos = [2, 253, 4, 41, 60, 254, 255]
 
         if self.is_active_active_dualtor:
             # Skip ICMP for active-active dualtor as it is duplicated to both ToRs

--- a/ansible/roles/test/files/ptftests/py3/hash_test.py
+++ b/ansible/roles/test/files/ptftests/py3/hash_test.py
@@ -242,8 +242,7 @@ class HashTest(BaseTest):
         # ip_proto 60 is redirected to ip_proto 4 as encapsulation protocol, ip payload will be malformat
         # ip_proto 254 is experimental
         # MLNX ASIC can't forward ip_proto 254, BRCM is OK, skip for all for simplicity
-        # ip_proto 255 is reserved
-        skip_protos = [2, 253, 4, 41, 60, 254, 255]
+        skip_protos = [2, 253, 4, 41, 60, 254]
 
         if self.is_active_active_dualtor:
             # Skip ICMP for active-active dualtor as it is duplicated to both ToRs
@@ -254,6 +253,8 @@ class HashTest(BaseTest):
             skip_protos.append(0)
             # Skip IPv6-ICMP for active-active dualtor as it is duplicated to both ToRs
             skip_protos.append(58)
+            # next-header 255 on BRCM causes 4 bytes to be stripped (CS00012366805)
+            skip_protos.append(255)
 
         while True:
             ip_proto = random.randint(0, 255)

